src/vampir/vampir.lisp

```diff
@@ -68,3 +68,256 @@
 (defun next-range (range x)
   (make-application :func :next_range
                     :arguments (list range x)))
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+;; Range
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+(defparameter *range32*
+  (labels ((in-plus (n)
+             (if (= n 0)
+                 (make-wire :var :a0)
+                 (make-infix :op :+
+                             :lhs (in-plus (- n 1))
+                             :rhs (make-infix :op :*
+                                              :lhs (make-constant :const (expt 2 n))
+                                              :rhs (make-wire :var (intern
+                                                                    (format nil "A~a" n) :keyword)))))))
+    (labels ((in-tuple (n)
+               (if (= n 0)
+                   (make-wire :var :a0)
+                   (make-tuples :wires (list
+                                        (in-tuple (- n 1))
+                                        (make-wire :var (intern
+                                                         (format nil "A~a" n) :keyword)))))))
+      (make-alias
+       :name :range32
+       :inputs (list :a)
+       :body
+       (append (mapcar
+                (lambda (x y)
+                  (spc:make-bind :names (list (make-wire :var y)) :value
+                                 (make-application :func :fresh
+                                                   :arguments (list (make-constant :const x)))))
+                (mapcar (lambda (x) (expt 2 x))
+                        (alexandria:iota 32 :start 1))
+                (mapcar (lambda (x) (intern
+                                     (format nil "A~a" x) :keyword))
```

Comment:
best not to indent on the same line as a `lambda`. Since many people's configurations change lambda to `λ` 

```suggestion
                (mapcar (lambda (x)
                          (intern (format nil "A~a" x) :keyword))
```